vortex-buffer/src/lib.rs

```diff
@@ -121,6 +121,20 @@ impl Buffer {
     }
 }
 
+impl PartialEq for Buffer {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref().eq(other.as_ref())
```

Comment:
might be better to use `as_slice()` explicitly so that if we add another AsRef impl for this type it doesn't end up picking something unexpected